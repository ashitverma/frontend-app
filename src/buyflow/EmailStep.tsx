import React, { useState } from "react";

interface EmailStepProps {
  cb: (field: string, value: string) => void;
}

const EmailStep: React.FC<EmailStepProps> = (props) => {
  const [email, setEmail] = useState("");
  const validateEmail = (value: string) => {
    const pattern = new RegExp(/(.+)@(\w+)\.(\w+)$/);
    const validEmailValue = pattern.test(value);
    !validEmailValue && setEmail("");
    !validEmailValue && alert("please enter a valid email");
  };
  return (
    <>
      <div>
        <label htmlFor="email">Email : </label>
        <input
          type="email"
          onChange={({ target: { value } }) => {
            setEmail(value);
          }}
          value={email}
          onBlur={(e) => validateEmail(e.target.value)}
        ></input>
      </div>
      <button onClick={() => props.cb("email", email)}>Next</button>
    </>
  );
};

export default EmailStep;
