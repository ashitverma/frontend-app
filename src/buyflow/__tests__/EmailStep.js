import { render } from "@testing-library/react"; // (or /dom, /vue, ...)
import EmailStep from "../EmailStep";

test("renders a email step input field", () => {
  const queries = render(<EmailStep />);
  expect(queries.getByText(/email/i)).toBeInTheDocument();
});

it("should take a snapshot", () => {
  const { asFragment } = render(<EmailStep />);

  expect(asFragment(<EmailStep />)).toMatchSnapshot();
});
