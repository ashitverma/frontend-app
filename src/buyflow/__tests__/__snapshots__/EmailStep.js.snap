// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <label
      for="email"
    >
      Email:
    </label>
    <input
      type="email"
      value=""
    />
  </div>
  <button
    disabled=""
  >
    Next
  </button>
</DocumentFragment>
`;
